pkg/api/agentPoolOnlyApi/v20180331/validate.go

```diff
@@ -37,9 +37,6 @@ func (a *AgentPoolProfile) Validate() error {
 func (l *LinuxProfile) Validate() error {
 	// Don't need to call validate.Struct(l)
 	// It is handled by Properties.Validate()
-	if e := validate.Var(l.SSH.PublicKeys[0].KeyData, "required"); e != nil {
```

Comment:
and here

Reply:
See previous. It's not possible to get here with an empty string, without a fatal error. So it's not something we need to validate on the regular.

Reply:
same comment, this one is also for AKS, we don't want to make any changes there (they don't use generate/deploy).